File: analyze_excel.py
import pandas as pd
import json

try:
    file_path = r"c:\Users\bilij\Documents\iiit\rkvalley.xlsx"
    # Read the first few rows to understand structure
    df = pd.read_excel(file_path, nrows=5)
    
    print("Columns:", df.columns.tolist())
    print("\nFirst 5 rows:")
    print(df.to_string())
    
except Exception as e:
    print(f"Error reading Excel file: {e}")

File: analyze_excel.js
import readXlsxFile from 'read-excel-file/node';

const schema = {
    'Student ID': {
        prop: 'id',
        type: String
    },
    'Name': {
        prop: 'name',
        type: String
    },
    'Class': {
        prop: 'classSection',
        type: String
    },
    'Email': {
        prop: 'email',
        type: String
    }
}

readXlsxFile('C:\\Users\\bilij\\Documents\\iiit\\TIMETABLE.xlsx').then((rows) => {
    console.log("Total rows:", rows.length);
    console.log("Headers:", rows[0]);
    console.log("First 5 data rows:");
    rows.slice(1, 6).forEach(row => console.log(row));
}).catch((error) => {
    console.error("Error reading Excel:", error);
});
